### ISSUE TITLE
Support for SSL Context Options

### ISSUE THREAD

**Issue by jzaeske** [WsdlToPhp/PackageGenerator#40]
Hi,

thank you for maintaining this project. It's very useful.

Would it be possible to provide additional Options for the SSL Context like verify_peer, ca_file and ca_path mentioned in http://php.net/manual/de/context.ssl.php.
Currently I have some problems to generate Code for WSDLs which are hosted under https with self-signed certificates or with CAs which are not in my trust-chain

**Comment by mikaelcom** [comment]
Hi,

I must admit this is not documented so could you try using the Soap Options that can be customized using the GeneratorOptions beginning with this [doc section](https://github.com/WsdlToPhp/PackageGenerator#dealing-with-the-options) then call such as (if I'm  not wrong):
```php
// ensure I did not a mistake here ;)
$context = stream_context_create(
    array(
        'ssl' => 
            array(
                'verify_peer' => '',
                'ca_file' => '',
                'ca_path' => ''
            )
    )
);
// the options
$options = GeneratorOptions::instance(/* '/path/to/your/configuration/file.yml' */);
$options->.....
// then add
$options->setSoapOptions(
    array(
        \WsdlToPhp\PackageBase\SoapClientInterface::WSDL_STREAM_CONTEXT => $context
    )
);
// then ...
$generator = new Generator($options);
$generator->generatePackage();
```

Get back to me if this is not clear :wink:, in any case in fact in order to know if it works this way,

Then I'll see if this is something that can be added as options in the command line tool

Good luck,

**Comment by jzaeske** [comment]
Hi,

thanks for the quick response. I've tried to use setSoapOptions, but it looks like the Soap Options are not considered when the problem appears.

The error occurs when first accessing the origin url with `file_get_contents()` in this call stack:
```
#1 .../wsdltophp/packagegenerator/src/Generator/Utils.php(80): file_get_contents('<url>', false, NULL)
#2 .../wsdltophp/packagegenerator/src/Generator/Generator.php(789): WsdlToPhp\PackageGenerator\Generator\Utils::getContentFromUrl('<url>', '', '', '', '', '', '')
#3 .../wsdltophp/packagegenerator/src/Generator/Generator.php(135): WsdlToPhp\PackageGenerator\Generator\Generator->getUrlContent('<url>')
```

The function `Utils:getContentFromUrlContextOptions()` creates a context for this requests, considering options for proxy and http auth, but not ssl verifcation.

Regards.

**Comment by mikaelcom** [comment]
Indeed, a whole new context is created to download the WSDL and its schemas (if there is any). I think it would only require to take into account the context passed as parameter to the SoapClient in order to be fully functionnal. I'll see what I can ASAP,

Jus to now, is the way I suggested to you easy and comprehensible enough? I think I'll have to add it to the readme or the FAQ, let me know thx

**Comment by jzaeske** [comment]
>  I think it would only require to take into account the context passed as parameter to the SoapClient in order to be fully functionnal.

I think that will be sufficient. This solution would be comprehensible enough for me. Thank you.

**Comment by mikaelcom** [comment]
just to let you know that I don't forget your issue and that I'll be able to work on it from February 1st :wink:

**Comment by mikaelcom** [comment]
@jzaeske Hi, could you try with the feature/issue-40 branch? Don't forget to do a `composer update` in order to get the last version of PackageBase.
Let me know then,
Thanks,

**Comment by jzaeske** [comment]
Hi @mikaelcom . Sorry for the late response.
I put a comment on the commit https://github.com/WsdlToPhp/PackageGenerator/commit/b26607a2fff4a9227cbac76cc42d899cd5cec076#commitcomment-15974412

**Comment by mikaelcom** [comment]
fixed in release [1.3.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/1.3.0)
